lib/gitarro/backend.rb

```diff
@@ -99,8 +99,42 @@ def run_script
     $CHILD_STATUS.exitstatus.nonzero? ? 'failure' : 'success'
   end
 
+  def export_pr_data(pr)
+      export_pr_data_to_simple_file(pr)
+      export_pr_data_to_file(pr)
+  end
+
   private
 
+  def export_pr_data_to_simple_file(pr)
+    # save this file in local dir where gitarro is executed.
+    # This part is kept for compatibility purposes
+    File.open('.gitarro_vars', 'w') do |file|
+       file.write("GITARRO_PR_AUTHOR: #{pr.head.user.login}\n" \
+       "GITARRO_PR_TITLE:  #{pr.title}\n" \
+       "GITARRO_PR_NUMBER: #{pr.number}\n" \
+       "GITARRO_PR_TARGET_REPO: #{@repo}\n")
+    end
+  end
+
+  def export_pr_data_to_file(pr)
+    pr = pr.to_hash
+    pr[:files] = []
+    @client.pull_request_files(@repo, pr[:number]).each do |ghfile|
```

Comment:
pr[:files] = @client.pull_request_files(@repo, pr[:number]).map { |ghfile| (ghfile.to_hash }

Reply:
i think this should work

Reply:
https://mixandgo.com/learn/how-to-use-the-ruby-map-method

Reply:
Mmm, I don't really understand this, and the syntax is invalid (there's a missing parentheses)

Can you explain?